junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,544 @@
+       * }
+       * }}}
+       */
+      def mkInvokeJUnitMethodOnModuleDef(methods: List[MethodSymbol],
+          bootSym: Symbol, mlClassSym: Option[Symbol]): DefDef = {
+
+        val invokeJUnitMethodSym = bootSym.newMethod(newTermName("invoke"))
+
+        val paramSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("methodName"))
+        paramSym.setInfo(typeOf[java.lang.String])
+
+        invokeJUnitMethodSym.setInfo(
+          MethodType(List(paramSym), definitions.UnitTpe))
+
+        val methodNotFound = {
+          val msg = gen.mkMethodCall(paramSym, nme.PLUS, Nil,
+            List(Literal(Constant(" not found"))))
+          val exception = mkNewInstance[NoSuchMethodException](List(msg))
+          Throw(exception)
+        }
+
+        val invokeJUnitMethodRhs0 = {
+          methods.foldRight[Tree](methodNotFound) {
+            case (methodSymbol, acc) =>
+              val methodSymbolLocal = {
+                mlClassSym.fold[Symbol](methodSymbol) { sym =>
+                  methodSymbol.cloneSymbol(newOwner = sym)
+                }
+              }
+              val headMethodId = Literal(Constant(methodSymbol.nameString))
+              If(gen.mkMethodCall(paramSym, nme.equals_, Nil,
+                List(headMethodId)), gen.mkMethodCall(methodSymbolLocal, Nil),acc)
+          }
+        }
+        val invokeJUnitMethodRhs =
+          atOwner(invokeJUnitMethodSym)(typer.typed(invokeJUnitMethodRhs0))
+        val param = newValDef_compat(paramSym, EmptyTree)()
+        typer.typedDefDef(newDefDef_compat(invokeJUnitMethodSym, invokeJUnitMethodRhs)(
+            vparamss = List(List(param))))
+      }
+
+      /**
+       * This method generates a method that invokes a test method in the class
+       * given its name. These methods have no parameters.
+       *
+       * Example:
+       * {{{
+       * class Foo {
+       *   @Test def bar(): Unit
+       *   @Test def baz(): Unit
+       * }
+       * object Foo\$scalajs\$junit\$bootstrapper
+       *   // This is the method generated by mkInvokeJUnitMethodOnInstanceDef
+       *   def invoke(instance: AnyRef, methodName: String): Unit = {
+       *     if (methodName == "bar") instance.asInstanceOf[Foo].bar()
+       *     else if (methodName == "baz") instance.asInstanceOf[Foo].baz()
+       *     else throw new NoSuchMethodException(methodName + " not found")
+       *   }
+       * }
+       * }}}
+       */
+      def mkInvokeJUnitMethodOnInstanceDef(methods: List[MethodSymbol],
+          classSym: Symbol, refClassSym: Symbol): DefDef = {
+        val invokeJUnitMethodSym = classSym.newMethod(newTermName("invoke"))
+
+        val instanceParamSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("instance"))
+        instanceParamSym.setInfo(definitions.ObjectTpe)
+
+        val idParamSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("methodName"))
+        idParamSym.setInfo(definitions.StringTpe)
```

Comment:
`StringTpe` does not exist in Scala 2.10. You'll have to use `StringClass.toType`.